src/printer.js

```diff
@@ -398,10 +407,20 @@ function expressionParser(text, parsers, opts) {
 
 /** @type {import('prettier').Printer['embed']} */
 function embed(path, print, textToDoc, opts) {
+  if (!opts.__astro) opts.__astro = {};
+
   const node = path.getValue();
 
   if (!node) return null;
 
+  if (node.__isRawMarkdown) {
+    return textToDoc(getUnencodedText(node), { ...opts, parser: 'markdown' });
```

Comment:
The tricky part is detecting that the `<Markdown>` component contains children that are markdown itself. But once we assert that, and flag the node, we can easily reroute it to Prettier’s `markdown` parser.

Reply:
Oh hm. Yeah we probably do need to write a test for Astro components within markdown—you‘re right.

Reply:
This PR does improve markdown formatting in general, and since we have a little more work to do with Astro components in Markdown, maybe we can wait until that improves and then change this? This might have to happen when this Prettier plugin switches over to the new compiler (which we probably won’t start working on until the new compiler is ready for release)

Reply:
All that to say, open to feedback! But maybe we can keep this for now, and reverse it later if this diverges heavily from Astro usage

Reply:
I suppose we'd have fun essentially writing our own markdown x astro formatter if we did support components in Markdown.